ethereum/eip712/eip712.go

```diff
@@ -363,6 +355,55 @@ func jsonNameFromTag(tag reflect.StructTag) string {
 	return parts[0]
 }
 
+func unpackIfAny(
+	cdc codectypes.AnyUnpacker,
+	_fieldType reflect.Type,
+	_field reflect.Value,
+	assumeMsgStructure bool,
+) (reflect.Type, reflect.Value, error) {
+	// Verify type is Any, else return as-is
+	if _fieldType != cosmosAnyType {
```

Comment:
we should move this outside the function (to `traverseFields`) and only call it when field type is `Any`